commands/new_function.go

```diff
@@ -223,7 +226,8 @@ func prepareYAMLContent(appendMode bool, gateway string, function *stack.Functio
 `
 	if !appendMode {
 
-		yamlContent = `provider:
+		yamlContent = `version: ` + apiVersion + `
```

Comment:
I'm surprised that the unit tests didn't fail when you made this change.

If they passed then that's not great, but please find the test you need and make sure it now looks for a version.